### PR TITLE
Feature/search functionality

### LINKED ISSUE #7: User Story : Search Functionality
**As a user,**
**When I  ,** visit the community board
**I want to  ,** be able to search for specific produce near my location
**So that** I can find someone in my area who has extra produce to share.

**Scenario:** A user navigates to the community board to find available produce in their area. 

**Given that ...** the user is on the community board, they have the ability to search all posts for a specific produce type in order to find another user that has produce available to share.

### PR DESCRIPTION
**Pull request by christina-cook**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
Added search functionality to the search bar. 

## *Where should the reviewer start?*
Bug is occurring in line 39 of `Feed.js`.

## *What are the relevant tickets?*
Closes #7

## *Concerns / Bugs:*
The home view is working as expected and rendering posts based on the search, however there's a glitch when trying to navigate to the Profile page. It has to do with the `searchResults` array, which has no length on the Profile page. In need of a second set of eyes on this one.

## Checklist:
- [ ] I have performed a self-review of my own code
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/views/ProfileView/ProfileView.js [hunk 1]
```diff
@@ -20,7 +21,7 @@ const ProfileView = () => {
       <div className='feed-container'>
         <UserInteractions />
         <h3 className='feed-title'>Posts</h3>
-        <Feed profileIsVisible={profileIsVisible}/>
+        <Feed profileIsVisible={profileIsVisible} searchResults={searchResults} />
```
Comment (christina-cook) [inline src/views/ProfileView/ProfileView.js:24]: I assume this fixed the error on the profile page? Seems so simple. Gosh, I was looking at this code for far too long.

File: src/components/Search/Search.js [hunk 2]
```diff
@@ -32,6 +32,7 @@ const Search = ({ determineSearchResults }) => {
     <form className='search' onSubmit={handleSubmit}>
       <input
         type='text'
+        value={searchInput}
```
Comment (christina-cook) [inline src/components/Search/Search.js:35]: Not sure how I missed this but thank you for catching that!

File: src/views/ProfileView/ProfileView.js (hunk 1 above)
Comment (colefiscus) [reply src/views/ProfileView/ProfileView.js:24]: Yeah that seemed to do the trick haha.

File: src/components/Header/Header.js [hunk 3]
```diff
@@ -5,20 +5,18 @@ import { Link } from 'react-router-dom';
...
+    homeLink.classList.toggle('hidden')
+    profileLink.classList.toggle('hidden')
   }
 
   return (
     <header className='header'>
       <h1 className='veg-share-title'>Garden Party</h1>
       <div className='header-links'>
-        <button className='home-link link-button hidden'>
+        <button className='link-button'>
```
Comment (colefiscus) [inline src/components/Header/Header.js:18]: This was my code, but if there's time we may want to improve this - button only works right now if you click on the text (the Link). Tried to just widen the Link, but wasn't cooperating.

**Review by colefiscus — APPROVED** [review]
Great work adding the Search and the 'clear search' button along with the styling and functionality! Glad we got it working right.

One possible future improvement is to create a different message to appear if there are no search results. I got a message to show up, but it's now the same one that shows up if there's no existing data. So some differentiation would be good.

File: src/components/Header/Header.js (hunk 3 above)
Comment (christina-cook) [reply src/components/Header/Header.js:18]: Hmm, that is weird. We can definitely look into fixing that later today or even tomorrow.

**Comment by christina-cook** [comment]
Thanks for helping figure out that error I was dealing with Cole! I agree that it would be good to have a different message if the search doesn't return any posts.

File: src/components/Search/Search.js (hunk 2 above)
Comment (dietza) [reply src/components/Search/Search.js:39]: Yep, I'd fixed this on my branch too and talked through it here with Cole, so this should help with those state updates 🤘

File: src/components/Header/Header.js (hunk 3 above)
Comment (dietza) [reply src/components/Header/Header.js:18]: Yeah that's weird, I wonder if wrapping in a link would help?

**Comment by dietza** [comment]
> Great work adding the Search and the 'clear search' button along with the styling and functionality! Glad we got it working right.
> 
> One possible future improvement is to create a different message to appear if there are no search results. I got a message to show up, but it's now the same one that shows up if there's no existing data. So some differentiation would be good.

Like that idea for handling an empty search result!
Agreed - thanks for the awesome troubleshooting to resolve that search bug 👏

**Review by dietza — APPROVED** [review]

**Pull request merged**